src/modules/CreateDrops/NftDrop/CreateNftDropForm.tsx

```diff
@@ -0,0 +1,98 @@
+import { useCallback } from 'react';
+import { Controller, useFormContext } from 'react-hook-form';
+import { Button, Flex, Input, VStack } from '@chakra-ui/react';
+
+import { IconBox } from '@/common/components/IconBox';
+import { LinkIcon } from '@/common/components/Icons';
+import { FormControl } from '@/common/components/FormControl';
+import { Checkboxes } from '@/common/components/Checkboxes';
+
+import { useDropFlowContext } from '../contexts/DropFlowContext';
+
+import { WALLET_OPTIONS } from './CREATE_NFT_DROP_TESTDATA';
+import { NftNameInput } from './Fields/NftNameInput';
+import { DescriptionInput } from './Fields/DescriptionInput';
+import { ArtworkInput } from './Fields/ArtworkInput';
+
+export interface CreateNftDropFormFieldTypes {
+  nftName: string;
+  description: string;
+  artwork: string;
+  selectedToWallets: string[];
+  redirectLink?: string;
+}
+
+export const CreateNftDropForm = () => {
+  const { onNext } = useDropFlowContext();
+  const {
+    setValue,
+    handleSubmit,
+    control,
+    formState: { isDirty, isValid },
+  } = useFormContext<CreateNftDropFormFieldTypes>();
+
+  const handleCheckboxChange = useCallback(
+    (value) => {
+      setValue('selectedToWallets', value, { shouldValidate: true });
+    },
+    [setValue],
+  );
+
+  const handleSubmitClick = () => {
+    onNext();
+  };
+
+  return (
+    <IconBox icon={<LinkIcon />} maxW={{ base: '21.5rem', md: '36rem' }} mx="auto">
+      <form onSubmit={handleSubmit(handleSubmitClick)}>
+        <VStack spacing={{ base: '4', md: '5' }}>
+          <NftNameInput control={control} />
```

Comment:
just curious, why did you separate out the text inputs?

Reply:
sorry what do you mean?

Reply:
i meant why not just put `<Controller ...` here in this file instead of separating it out to `<NftNameInput` if that makes sense